src/pages/studyView/tabs/CNSegments.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react';
+import {observer} from "mobx-react";
+import {action, computed, observable} from "mobx";
+import autobind from "autobind-decorator";
+
+import {CopyNumberSeg} from "shared/api/generated/CBioPortalAPI";
+import IntegrativeGenomicsViewer from "shared/components/igv/IntegrativeGenomicsViewer";
+import {calcSegmentTrackHeight, defaultSegmentTrackProps, generateSegmentFeatures} from "shared/lib/IGVUtils";
+import ProgressIndicator, {IProgressIndicatorItem} from "shared/components/progressIndicator/ProgressIndicator";
+import LoadingIndicator from "shared/components/loadingIndicator/LoadingIndicator";
+import CNSegmentsDownloader from "shared/components/cnSegments/CNSegmentsDownloader";
+
+import {StudyViewPageStore} from "../StudyViewPageStore";
+
+@observer
+export default class CNSegments extends React.Component<{ store: StudyViewPageStore }, {}> {
+    @observable renderingComplete = false;
+
+    @computed get segmentTrackHeight() {
+        return calcSegmentTrackHeight(this.features);
+    }
+
+    @computed get features() {
+        const segments: CopyNumberSeg[] = this.activePromise ? this.activePromise.result || [] : [];
+
+        return generateSegmentFeatures(segments);
+    }
+
+    @computed get filename()
+    {
+        let prefix = "";
+
+        if (this.props.store.studyIds.length > 0) {
+            if (this.props.store.studyIds.length > 1) {
+                prefix = "multi_study_";
+            }
+            else {
+                prefix = `${this.props.store.studyIds[0]}_`;
+            }
+        }
+
+        return `${prefix}segments.seg`;
+    }
+
+    @computed get isHidden() {
+        return this.isLoading || !this.renderingComplete;
+    }
+
+    @computed get isLoading()
+    {
+        return this.activePromise ? this.activePromise.isPending : true;
+    }
+
+    @computed get activePromise() {
+        return this.props.store.cnSegments;
+    }
+
+    @computed get progressItems(): IProgressIndicatorItem[] {
+        return [
+            {
+                label: "Downloading copy number segments data...",
```

Comment:
Let's use `Loading` instead of `Downloading` to keep the consistency with other tabs.